pkg/observ/tracing.go

```diff
@@ -0,0 +1,89 @@
+package observ
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/gobuffalo/buffalo"
+	"github.com/gomods/athens/pkg/config/env"
+	"github.com/gomods/athens/pkg/errors"
+	"go.opencensus.io/exporter/jaeger"
+	"go.opencensus.io/trace"
+)
+
+// ENV is used to define the sampling rate
+var ENV = env.GoEnvironmentWithDefault("development")
+
+// observabilityContext is a private context that is used by the packages to start the span
+type observabilityContext struct {
+	buffalo.Context
+	spanCtx context.Context
+}
+
+// RegisterTraceExporter returns a jaeger exporter for exporting traces to opencensus.
+// It should in the future have a nice sampling rate defined
+// TODO: Extend beyond jaeger
+func RegisterTraceExporter(service string) (*(jaeger.Exporter), error) {
+	const op errors.Op = "RegisterTracer"
+	collectorEndpointURI := env.TraceExporterURL()
+	if collectorEndpointURI == "" {
+		return nil, errors.E(op, "Exporter URL is empty. Traces won't be exported")
```

Comment:
let's silently fail here if we are in development. The sane default should have no tracing